File: MapDemo/Objective C/CovidSafetyStatusExample.m
//
//  CovidSafetyStatusExample.m
//  MapDemo
//
//  Created by apple on 18/06/20.
//  Copyright © 2020 MMI. All rights reserved.
//

#import "CovidSafetyStatusExample.h"

@interface CovidSafetyStatusExample () <MapmyIndiaMapViewDelegate>

@property (nonatomic) IBOutlet MapmyIndiaMapView *mapView;

@property (weak, nonatomic) IBOutlet UIButton *getCurrentSafetyButton;

@property (weak, nonatomic) IBOutlet UIButton *hideSafetyStatusButton;


@end

@implementation CovidSafetyStatusExample

BOOL isLocationReady = NO;

- (IBAction)getCurrentSafetyButtonPressed:(UIButton *)sender {
    [_mapView showCurrentLocationSafety];
}

- (IBAction)hideSafetyStatusButtonPressed:(UIButton *)sender {
    [_mapView hideSafetyStrip];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    _mapView.showsUserLocation = YES;
    _mapView.userTrackingMode = MGLUserTrackingModeFollow;
    _mapView.delegate = self;
}

- (void)mapView:(MGLMapView *)mapView didUpdateUserLocation:(MGLUserLocation *)userLocation {
    if (userLocation && userLocation.location && CLLocationCoordinate2DIsValid(userLocation.location.coordinate)) {
        isLocationReady = YES;
    }
}

@end

File: MapDemo/Swift/CovidSafetyStatusExample.swift
//
//  CovidSafetyStatusExample.swift
//  MapDemo
//
//  Created by apple on 18/06/20.
//  Copyright © 2020 MMI. All rights reserved.
//

import UIKit
import MapmyIndiaMaps

@objc(CovidSafetyStatusExample_Swift)
class CovidSafetyStatusExample_Swift: UIViewController {

    @IBOutlet weak var mapView: MapmyIndiaMapView!
    
    @IBOutlet weak var getCurrentSafetyButton: UIButton!
    
    @IBOutlet weak var hideSafetyStatusButton: UIButton!
    
    @IBAction func getCurrentSafetyButtonPressed(_ sender: UIButton) {
        if isLocationReady {
            mapView.showCurrentLocationSafety()
        }
    }
    
    @IBAction func hideSafetyStatusButtonPressed(_ sender: UIButton) {
        mapView.hideSafetyStrip()
    }
        
    var isLocationReady = false
    
    override func viewDidLoad() {
        super.viewDidLoad()

        mapView.delegate = self
        mapView.showsUserLocation = true
        mapView.userTrackingMode = .follow
    }

}

extension CovidSafetyStatusExample_Swift: MapmyIndiaMapViewDelegate {
    func mapView(_ mapView: MGLMapView, didUpdate userLocation: MGLUserLocation?) {
        if let userLocation = userLocation, let location = userLocation.location, CLLocationCoordinate2DIsValid(location.coordinate) {
            isLocationReady = true
        }
    }
}
